tests/kafkatest/services/kafka.py

```diff
@@ -100,6 +106,16 @@ def signal_leader(self, topic, partition=0, sig=signal.SIGTERM):
         leader = self.leader(topic, partition)
         self.signal_node(leader, sig)
 
+    def stop(self):
```

Comment:
You can remove this `stop` method:
The act of instantiating a `Service`, in this case `minikdc`, registers the service with ducktape, and ducktape `stops` and `cleans` every registered service at the end of the test. 

Since `MiniKdc` has defined its `stop_node` and `clean_node` methods, ducktape will clean it up after the test has run.